Synthesis failed for nodejs-resource

Hello! Autosynth couldn't regenerate nodejs-resource. :broken_heart:

Here's the output from running `synth.py`:

```
Cloning into 'working_repo'...
Switched to a new branch 'autosynth'
Cloning into '/tmpfs/tmp/tmppabks13v/synthtool'...
Switched to branch 'autosynth-self'
Note: checking out 'ed2b39c2dba2fa61a2c7efffc9a774881ca2d0cb'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at ed2b39c chore(deps): update dependency uuid to v8 (#315)
Note: checking out '19465d3ec5e5acdb01521d8f3bddd311bcbee28d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 19465d3 build: use codecov's action, now that it's authless (#499)
Switched to a new branch 'autosynth-self-1'
2020-05-06 06:06:36 [INFO] Running synthtool
2020-05-06 06:06:36 [INFO] ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-06 06:06:36,487 synthtool > Executing /tmpfs/src/github/synthtool/working_repo/synth.py.
On branch autosynth-self-1
nothing to commit, working tree clean
2020-05-06 06:06:36,620 synthtool > Using precloned repo /tmpfs/tmp/tmppabks13v/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-06 06:06:36,749 synthtool > Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/resource@1.1.5 prepare /tmpfs/src/github/synthtool/working_repo
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/resource@1.1.5 precompile /tmpfs/src/github/synthtool/working_repo
> gts clean

Removing build ...

> @google-cloud/resource@1.1.5 compile /tmpfs/src/github/synthtool/working_repo
> tsc -p .

[96mtest/index.ts[0m:[93m142[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'baseUrl' does not exist on type '{}'.

[7m142[0m       assert.strictEqual(calledWith.baseUrl, baseUrl);
[7m   [0m [91m                                    ~~~~~~~[0m

[96mtest/index.ts[0m:[93m143[0m:[93m41[0m - [91merror[0m[90m TS2339: [0mProperty 'scopes' does not exist on type '{}'.

[7m143[0m       assert.deepStrictEqual(calledWith.scopes, [
[7m   [0m [91m                                        ~~~~~~[0m

[96mtest/index.ts[0m:[93m146[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'projectIdRequired' does not exist on type '{}'.

[7m146[0m       assert.strictEqual(calledWith.projectIdRequired, false);
[7m   [0m [91m                                    ~~~~~~~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m148[0m:[93m20[0m - [91merror[0m[90m TS2339: [0mProperty 'packageJson' does not exist on type '{}'.

[7m148[0m         calledWith.packageJson,
[7m   [0m [91m                   ~~~~~~~~~~~[0m

[96mtest/project.ts[0m:[93m85[0m:[93m34[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m85[0m       const calledWith = project.calledWith_[0];
[7m  [0m [91m                                 ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/resource@1.1.5 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/resource@1.1.5 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T13_06_52_617Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/resource@1.1.5 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/resource@1.1.5 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T13_06_52_656Z-debug.log
2020-05-06 06:06:52,685 synthtool > Failed executing npm install:

None
2020-05-06 06:06:52,686 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmpfs/src/github/synthtool/working_repo/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 166, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-06 06:06:52 [ERROR] Synthesis failed
HEAD is now at ed2b39c chore(deps): update dependency uuid to v8 (#315)
Switched to branch 'autosynth-self'
Note: checking out 'ed2b39c2dba2fa61a2c7efffc9a774881ca2d0cb'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at ed2b39c chore(deps): update dependency uuid to v8 (#315)
Previous HEAD position was 19465d3 build: use codecov's action, now that it's authless (#499)
HEAD is now at ab88356 fix: make .kokoro-autosynth executable (#522)
Switched to a new branch 'autosynth-17'
2020-05-06 06:06:52 [INFO] Running synthtool
2020-05-06 06:06:52 [INFO] ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-06 06:06:52,978 synthtool > Executing /tmpfs/src/github/synthtool/working_repo/synth.py.
On branch autosynth-17
nothing to commit, working tree clean
2020-05-06 06:06:53,110 synthtool > Using precloned repo /tmpfs/tmp/tmppabks13v/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-06 06:06:53,235 synthtool > Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/resource@1.1.5 prepare /tmpfs/src/github/synthtool/working_repo
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/resource@1.1.5 precompile /tmpfs/src/github/synthtool/working_repo
> gts clean

Removing build ...

> @google-cloud/resource@1.1.5 compile /tmpfs/src/github/synthtool/working_repo
> tsc -p .

[96mtest/index.ts[0m:[93m142[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'baseUrl' does not exist on type '{}'.

[7m142[0m       assert.strictEqual(calledWith.baseUrl, baseUrl);
[7m   [0m [91m                                    ~~~~~~~[0m

[96mtest/index.ts[0m:[93m143[0m:[93m41[0m - [91merror[0m[90m TS2339: [0mProperty 'scopes' does not exist on type '{}'.

[7m143[0m       assert.deepStrictEqual(calledWith.scopes, [
[7m   [0m [91m                                        ~~~~~~[0m

[96mtest/index.ts[0m:[93m146[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'projectIdRequired' does not exist on type '{}'.

[7m146[0m       assert.strictEqual(calledWith.projectIdRequired, false);
[7m   [0m [91m                                    ~~~~~~~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m148[0m:[93m20[0m - [91merror[0m[90m TS2339: [0mProperty 'packageJson' does not exist on type '{}'.

[7m148[0m         calledWith.packageJson,
[7m   [0m [91m                   ~~~~~~~~~~~[0m

[96mtest/project.ts[0m:[93m85[0m:[93m34[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m85[0m       const calledWith = project.calledWith_[0];
[7m  [0m [91m                                 ~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/resource@1.1.5 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/resource@1.1.5 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T13_06_59_475Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/resource@1.1.5 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/resource@1.1.5 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T13_06_59_491Z-debug.log
2020-05-06 06:06:59,506 synthtool > Failed executing npm install:

None
2020-05-06 06:06:59,507 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmpfs/src/github/synthtool/working_repo/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 166, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-06 06:06:59 [ERROR] Synthesis failed
HEAD is now at ed2b39c chore(deps): update dependency uuid to v8 (#315)
Switched to branch 'autosynth'
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 584, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 465, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 574, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 360, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 370, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 115, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](https://sponge/d5e01c0b-0d2d-4da4-b495-86674ea6021f).

--- a/test/index.ts
+++ b/test/index.ts
@@ -137,7 +137,7 @@ describe('Resource', () => {
     it('should inherit from Service', () => {
       assert(resource instanceof FakeService);
 
-      const calledWith = resource.calledWith_[0];
+      const calledWith = (resource as any).calledWith_[0];
       const baseUrl = 'https://cloudresourcemanager.googleapis.com/v1';
       assert.strictEqual(calledWith.baseUrl, baseUrl);
       assert.deepStrictEqual(calledWith.scopes, [

--- a/test/project.ts
+++ b/test/project.ts
@@ -82,7 +82,7 @@ describe('Project', () => {
       const project = new Project(resourceInstance, ID);
       assert(project instanceof ServiceObject);
 
-      const calledWith = project.calledWith_[0];
+      const calledWith = (project as any).calledWith_[0];
 
       assert.strictEqual(calledWith.parent, resourceInstance);
       assert.strictEqual(calledWith.baseUrl, '/projects');

